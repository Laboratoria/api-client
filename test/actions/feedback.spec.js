import {
  addProjectFeedback,
  deleteProjectFeedback,
  getProjectFeedback,
  sendProjectFeedback,
  updateProjectFeedback,
} from '../../lib/actions/feedback';

describe('getProjectFeedback', () => {
  it('should be a function', () => {
    expect(typeof getProjectFeedback).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(getProjectFeedback({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
    })).toMatchSnapshot();
  });
});

describe('addProjectFeedback', () => {
  it('should be a function', () => {
    expect(typeof addProjectFeedback).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(addProjectFeedback({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
      data: { foo: 'bar' },
    })).toMatchSnapshot();
  });
});

describe('updateProjectFeedback', () => {
  it('should be a function', () => {
    expect(typeof updateProjectFeedback).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(updateProjectFeedback({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
      data: { foo: 'bar' },
    })).toMatchSnapshot();
  });
});

describe('deleteProjectFeedback', () => {
  it('should be a function', () => {
    expect(typeof deleteProjectFeedback).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(deleteProjectFeedback({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
    })).toMatchSnapshot();
  });
});

describe('sendProjectFeedback', () => {
  it('should be a function', () => {
    expect(typeof sendProjectFeedback).toBe('function');
  });

  it('should create an appropriate action', () => {
    expect(sendProjectFeedback({
      user: 'someone',
      cohortId: 'someone',
      projectId: 'someone',
    })).toMatchSnapshot();
  });
});
