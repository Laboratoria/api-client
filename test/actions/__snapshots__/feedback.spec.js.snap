// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addProjectFeedback should create an appropiate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-feedback-add",
    "namespace": "default",
    "suffix": "PROJECT_FEEDBACK_ADD",
  },
  "payload": Object {
    "data": Object {
      "foo": "bar",
    },
    "headers": undefined,
    "method": "POST",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/feedback",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_FEEDBACK_ADD",
}
`;

exports[`deleteProjectFeedback should create an appropiate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-feedback-delete",
    "namespace": "default",
    "suffix": "PROJECT_FEEDBACK_DELETE",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "DELETE",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/feedback",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_FEEDBACK_DELETE",
}
`;

exports[`getProjectFeedback should create an appropiate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "user/someone/feedback",
    "namespace": "default",
    "suffix": "PROJECT_FEEDBACK",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "GET",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/feedback",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_FEEDBACK",
}
`;

exports[`sendProjectFeedback should create an appropiate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-feedback-send",
    "namespace": "default",
    "suffix": "PROJECT_FEEDBACK_SEND",
  },
  "payload": Object {
    "data": undefined,
    "headers": undefined,
    "method": "POST",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/feedback/_send",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_FEEDBACK_SEND",
}
`;

exports[`updateProjectFeedback should create an appropiate action 1`] = `
Object {
  "meta": Object {
    "anonymous": undefined,
    "expiration": undefined,
    "key": "project-feedback-update",
    "namespace": "default",
    "suffix": "PROJECT_FEEDBACK_UPDATE",
  },
  "payload": Object {
    "data": Object {
      "foo": "bar",
    },
    "headers": undefined,
    "method": "PUT",
    "params": undefined,
    "url": "/users/someone/cohorts/someone/projects/someone/feedback",
  },
  "type": "@@api-client/API_REQUEST/PROJECT_FEEDBACK_UPDATE",
}
`;
